faces.py

```diff
@@ -150,11 +182,36 @@ def get_random_in_batch(cursor, current_user):
                           ON people.person_id = stints.person_id
                         INNER JOIN user_batch
                           ON stints.batch_id = user_batch.batch_id
-                      WHERE people.person_id != %s
+                      WHERE people.person_id != %s AND NOT image_url LIKE '%%no_photo_%%'
                       ORDER BY random()
-                      LIMIT 4""",
-                   [current_user,current_user])
-    return [x for x in cursor.fetchall()]
+                      LIMIT 1""",
+                   [current_user, current_user])
+
+    person = cursor.fetchall()
+
+    cursor.execute("""WITH user_batch AS (
+                        SELECT batch_id
+                        FROM stints
+                        WHERE person_id = %s
+                      )
+                      SELECT people.person_id,
+                             people.first_name,
+                             people.middle_name,
+                             people.last_name,
+                             people.image_url
+                      FROM people
+                        INNER JOIN stints
+                          ON people.person_id = stints.person_id
+                        INNER JOIN user_batch
+                          ON stints.batch_id = user_batch.batch_id
+                      WHERE people.person_id != %s AND people.person_id != %s
```

Comment:
We talked about this in person, too, and again it doesn't need to be fixed in this PR:

Now that we're doing two separate queries, and because these queries are each so long, I think it would make sense to extract a view. Possibly we'd also need to rename the existing `batch_mates` view, since that name might better describe this query.